core/frontend-devtools/src/effects/Explosion.ts

```diff
@@ -0,0 +1,211 @@
+
+  public static numEmissionsRange = Range1d.createXX(1, 5);
+
+  public constructor(texture: RenderTexture, iModel: IModelConnection, numEmissions: number) {
+    this._texture = texture;
+    this._pickableId = iModel.transientIds.next;
+    this._numEmissions = numEmissions;
+    this._lastUpdateTime = Date.now();
+
+    this._origin = randomPositionInRange(iModel.projectExtents);
+
+    this._dispose = iModel.onClose.addListener(() => this.dispose());
+  }
+
+  public dispose(): void {
+    if (this._dispose) {
+      this._dispose();
+      this._dispose = undefined;
+    }
+
+    IModelApp.viewManager.dropDecorator(this);
+    this._texture.dispose();
+  }
+
+  public update(): void {
+    const now = Date.now();
+    let deltaMillis = now - this._lastUpdateTime;
+    deltaMillis = Math.min(100, deltaMillis);
+    this._lastUpdateTime = now;
+
+    let numParticles = this._particles.length;
+    if (numParticles === 0) {
+      this._numEmissions--;
+      if (this._numEmissions < 0)
+        this.dispose();
+      else
+        this._particles = this._emitter.emit();
+
+      return;
+    }
+
+    const elapsedSeconds = deltaMillis / 1000;
+    for (let i = 0; i < numParticles; i++) {
+      const particle = this._particles[i];
+      this.updateParticle(particle, elapsedSeconds);
+      if (particle.isExpired) {
+        this._particles[i] = this._particles[numParticles - 1];
+        --i;
+        --numParticles;
+      }
+    }
+
+    this._particles.length = numParticles;
+  }
+
+  private updateParticle(particle: Particle, elapsedSeconds: number): void {
+    const velocity = particle.velocity.clone(this._scratchVector3d);
+    velocity.scale(elapsedSeconds, velocity);
+    velocity.z += elapsedSeconds * this.gravity;
+    particle.position.addInPlace(velocity);
+    particle.transparency = 255 * (particle.age / particle.lifetime);
+
+    particle.age += elapsedSeconds;
+  }
+
+  public decorate(context: DecorateContext): void {
+    if (!context.viewport.view.isSpatialView())
+      return;
+
+    this.update();
+
+    const builder = ParticleCollectionBuilder.create({
+      viewport: context.viewport,
+      texture: this._texture,
+      size: (this._emitter.sizeRange.high - this._emitter.sizeRange.low) / 2,
+      transparency: 0,
+      origin: this._origin,
+      pickableId: this._pickableId,
+    });
+
+    for (const particle of this._particles)
+      builder.addParticle(particle);
+
+    const graphic = builder.finish();
+    if (graphic) {
+      context.addDecoration(GraphicType.WorldDecoration, graphic);
+      context.viewport.onRender.addOnce((vp) => vp.invalidateDecorations());
+    }
+  }
+
+  public testDecorationHit(id: Id64String): boolean {
+    return id === this._pickableId;
+  }
+
+  public async getDecorationToolTip(_hit: HitDetail): Promise<HTMLElement | string> {
+    return "Explosion effect";
+  }
+
+  public async onDecorationButtonEvent(_hit: HitDetail, _ev: BeButtonEvent): Promise<EventHandled> {
```

Comment:
@bbastings I want to display a tooltip on mouseover, but I don't want the particle effect added to the selection set. This approximately does that - left-click on the thing won't select it, but box-select will. But if I start dragging for pan or rotate with the effect under the cursor, the view doesn't move. What is the right way to accomplish this?